### PR TITLE
New readme

### PR DESCRIPTION
**Pull request by ulysseB**
Improve the README and add a CONTRIBUTING file. Please comment to add suggestions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+The core Telamon library provides:
+- a program representation, in `src/ir`, to describe the operations we want to implement,
+- a search space representation, in `src/search_space`, to describe which optimisations
+  are available,
+- helpers to build program and search space instaces in `src/helper`,
+- search space exploration algorithms in `src/explorer` and
+- a performance model to help prune the search space in `src/model`.
+
+The search space is described in a custom language that exposes decisions and the
```
Comment (nicoTolly) [inline CONTRIBUTING.md]: typo : description

File: README.md [hunk 2]
```diff
@@ -2,73 +2,92 @@
...
-$ cargo build
+cargo bench --features=cuda --bench=cuda-search
 ```
-the tests are run with:
-```c
-$ cargo test
+in the `kernel/` directory. To see the progress of the exploration, append
+`RUST_LOG=telamon::explorer=warn` to the command.
+
+## Writting a Kernel
```
Comment (nicoTolly) [inline README.md]: Typo : Writing

File: README.md [hunk 3]
```diff
@@ -2,73 +2,92 @@
...
+
+To write a kernel, you must first define the inputs of the kernel and the context in which
```
Comment (nicoTolly) [inline README.md]: The context we optimize for ? The context in which we optimize ?

File: CONTRIBUTING.md (hunk 1 above)
Comment (ulysseB) [reply CONTRIBUTING.md]: Done

**Pull request merged**